src/vma/dev/ring_simple.cpp

```diff
@@ -162,21 +162,18 @@ ring_simple::~ring_simple()
 	// Was done in order to allow iperf's FIN packet to be sent.
 	usleep(25000);
 
-        /* coverity[double_lock] TODO: RM#1049980 */
-	m_lock_ring_rx.lock();
-	m_lock_ring_tx.lock();
-
 	if (m_p_qp_mgr) {
-		// 'down' the active QP/CQ
-		/* TODO: consider avoid using sleep */
-		/* coverity[sleep] */
-		m_p_qp_mgr->down();
+		stop_active_qp_mgr();
 
 		// Release QP/CQ resources
 		delete m_p_qp_mgr;
 		m_p_qp_mgr = NULL;
 	}
 
+	/* coverity[double_lock] TODO: RM#1049980 */
+	m_lock_ring_rx.lock();
```

Comment:
Consider changing locks order here or below, so the order is same through the code and the locks are unlocked in reverse order. (This is not related to current PR)